### PR TITLE
Minor Vent Crawl Buff; Hunters and Larva Can Vent Crawl Faster

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

1: Minor refactor of vent crawling. Some snowflaking removed. Infrastructure added for unique ventcrawl interactions.

2: Hunters can now crawl into and out of vents in 1 second down from 4.5 seconds.

3: Larva can now crawl into and out of vents in 1 second down from 4.5 seconds.

## Why It's Good For The Game

1. Removes snowflakery. 
2. Adds useful infrastructure for ventcrawl.
3. Castes that should be better at ventcrawling are.

## Changelog
:cl:
code: Partial refactor of vent crawling. Snowflaking removed. Infrastructure added for unique ventcrawl interactions, and custom ventcrawl parameters at the caste level.
balance: Hunters can now crawl into vents in 1 second down from 4.5 seconds.
balance: Larva can now crawl into vents in 1 second down from 4.5 seconds.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/xenoprocs.dm [hunk 1]
```diff
@@ -580,6 +580,27 @@
...
+// Larva Vent Crawl
+/mob/living/carbon/xenomorph/larva/vent_crawl()
+	set name = "Crawl through Vent"
+	set desc = "Enter an air vent and crawl through the pipe system."
+	set category = "Alien"
+	if(!check_state())
+		return
+	var/pipe = start_ventcrawl()
+	if(pipe)
+		handle_ventcrawl(pipe, LARVA_VENT_CRAWL_TIME, TRUE)
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:603]: maybe make vent crawl a var on xeno caste? 

this is a lot of duplicate code

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/xenoprocs.dm (hunk 1 above)
Comment (Surrealaser) [reply code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:603]: Not a bad idea, though there's already a headache inducing amount of vars which I hate adding to, but sure.

File: code/__DEFINES/mobs.dm [hunk 2]
```diff
@@ -528,7 +530,7 @@ GLOBAL_LIST_INIT(xenoupgradetiers, list(XENO_UPGRADE_BASETYPE, XENO_UPGRADE_INVA
 #define HUNTER_SILENCE_WINDUP 0.5 SECONDS //Windup of the Hunter's Silence
 #define HUNTER_SILENCE_MULTIPLIER 1.5 //Multiplier of stacks vs Hunter's Mark targets
 #define HUNTER_SILENCE_WHIFF_COOLDOWN 3 SECONDS //If we fail to target anyone with Silence, partial cooldown to prevent spam.
-
+#define HUNTER_VENT_CRAWL_TIME 1 SECONDS //Hunters can enter vents fast
```
Comment (TiviPlus) [inline code/__DEFINES/mobs.dm:533]: Make it 2 seconds for hunters

**Review by TiviPlus — COMMENTED** [review]
rest is fine

File: code/__DEFINES/mobs.dm (hunk 2 above)
Comment (Surrealaser) [reply code/__DEFINES/mobs.dm:533]: Eh, I tried it, seems pretty slow, and 1 is still enough of a delay that they can be interrupted/killed

Comment (Surrealaser) [reply code/__DEFINES/mobs.dm]: This is done

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**